cli/explain/explain.go

```diff
@@ -292,21 +292,25 @@ const (
 	oldOnlyState
 	newOnlyState
 	modifiedState
+	finalState
 )
 
 func writeSpawnDiffs(w io.Writer, diffs []*spawn_diff.SpawnDiff) {
 	// Diffs come in the order "old only", "new only", then "modified".
 	var oldOnly, newOnly map[string]uint32
 	previousState := initialState
-	for _, d := range diffs {
+	// Append a nil diff as a sentinel to ensure the final state is processed.
+	for _, d := range append(diffs, nil) {
```

Comment:
very nit: I wonder if there is a way to restructure this so that we would only need to check for `if i == len(diffs) {}` instead of appending the last element by ourselves.

Reply:
Good point. I that this would be somewhat errorprone as the last diff can already imply a state transition from the second to last diff. Handling another one in the same loop iteration would require more substantial changes.